packages/babel-helpers/src/helpers.js

```diff
@@ -601,6 +601,15 @@ helpers.set = () => template.program.ast`
 
       if (parent !== null) {
         _set(parent, property, value, receiver);
+      } else if (Object.defineProperty) {
+        Object.defineProperty(receiver, property, {
+          value: value,
+          enumerable: true,
+          configurable: true,
+          writable: true
+        });
```

Comment:
I don't understand why you're calling Object.defineProperty here. If the superclass prototype has a setter, it should be invoked.

Reply:
As far as I understand this code we are looking through the `prototype` chain and if we reach the end of that chain, then it means we haven't found the property nowhere so we are creating it.

Reply:
Oh sorry, I missed the recursive loop. Seems reasonable then.